### PR TITLE
[frameit] Skip warning message for already framed screenshots

### PR DESCRIPTION
**Pull request by KrauseFx**

### CODE REVIEW

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a changelog summary in the Pull Request description @KrauseFx</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by janpio** [comment]
```
[16:49:57]: ▸ Offenses:
[16:49:57]: ▸ frameit/lib/frameit/runner.rb:49:15: C: Metrics/BlockNesting: Avoid more than 3 levels of block nesting.
[16:49:57]: ▸ UI.error("Backtrace:\n\t#{ex.backtrace.join("\n\t")}") if FastlaneCore::Globals.verbose?
[16:49:57]: ▸ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[16:49:57]: ▸ 1024 files inspected, 1 offense detected
```

File: frameit/lib/frameit/runner.rb [hunk 1]
```diff
@@ -41,8 +41,13 @@ def run(path, color = nil)
             screenshot = Screenshot.new(full_path, color)
             screenshot.frame!
           rescue => ex
-            UI.error(ex.to_s)
-            UI.error("Backtrace:\n\t#{ex.backtrace.join("\n\t")}") if FastlaneCore::Globals.verbose?
+            # We don't want to show an error message for
+            #   [23:47:09]: Unsupported screen size [1295, 2590] for path './IMG_2343_framed_black.png'
+            # but instead silently skip framed screenshots
+            if full_path.ends_with?("_framed.png") && !FastlaneCore::Globals.verbose?
```
Comment (revolter) [inline frameit/lib/frameit/runner.rb]: Shouldn't this be `full_path.ends_with?("_framed.png")  !FastlaneCore::Globals.verbose?`

Comment (revolter) [inline frameit/lib/frameit/runner.rb]: Shouldn't this be `full_path.ends_with?("_framed.png") || !FastlaneCore::Globals.verbose?` instead?

**Comment by joshdholtz** [comment]
@KrauseFx Do you still want this merged? If so, I can fix the lint errors 😉

**Comment by KrauseFx** [comment]
@joshdholtz Yes I think this should still be merged, can you take this one over?

File: frameit/lib/frameit/runner.rb [hunk 2]
```diff
@@ -41,8 +41,13 @@ def run(path, color = nil)
             screenshot = Screenshot.new(full_path, color)
             screenshot.frame!
           rescue => ex
-            UI.error(ex.to_s)
-            UI.error("Backtrace:\n\t#{ex.backtrace.join("\n\t")}") if FastlaneCore::Globals.verbose?
+            # We don't want to show an error message for
+            #   [23:47:09]: Unsupported screen size [1295, 2590] for path './IMG_2343_framed_black.png'
```
Comment (janpio) [inline frameit/lib/frameit/runner.rb]: Isn't this an error message from `deliver`?
https://github.com/fastlane/fastlane/blob/c79cb094721dd2e0f49f4eb11aca36adf3f73666/deliver/lib/deliver/app_screenshot.rb#L304

We are already skipping files ending with `_framed.png` in line 29 of this file, so they don't get framed _again_. 

Is there an issue for this actual problem? My guess is that someone tried to upload a framed, but un-backgrounded screenshot (`frameit` was used to add a frame, but not a background/title/keyword) which doesn't resize the screenshot to App Store compatible sizes.

**Review by janpio — CHANGES_REQUESTED** [review]
I think this PR misses the problem it wants to solve - it should not be merged.

**Comment by KrauseFx** [comment]
Yep, good point, thanks @janpio 👍

**Pull request closed without merging**